test/jdk/javax/security/auth/callback/PasswordCallback/CheckCleanerBound.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2022 THL A29 Limited, a Tencent company. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8284910
+ * @summary Check that the cleaner is not bound to the PasswordCallback object
+ */
+
+import javax.security.auth.callback.PasswordCallback;
+import java.util.WeakHashMap;
+
+public final class CheckCleanerBound {
+    private final static WeakHashMap<PasswordCallback, ?> weakHashMap =
+            new WeakHashMap<>();
+
+    public static void main(String[] args) throws Exception {
+        // Create an object
+        PasswordCallback passwordCallback =
+                new PasswordCallback("Password: ", false);
+        passwordCallback.setPassword("ThisIsAPassword".toCharArray());
+
+        weakHashMap.put(passwordCallback, null);
+        passwordCallback = null;
+
+        // Check if the PasswordCallback object could be collected.
+        // Wait to trigger the cleanup.
+        for (int i = 0; i < 10 && weakHashMap.size() != 0; i++) {
+            System.gc();
+            Thread.sleep(100);
```

Comment:
You can drop this sleep to 10ms to cut the average test time.  It might be interesting to know how many retries are typical.

Reply:
Hm, it looks like a good idea to improve the testing performance (code updated).  The cleaner get called in the 1st GC collection on my local laptop.